autorest/adal/config.go

```diff
@@ -89,3 +93,58 @@ func NewOAuthConfigWithAPIVersion(activeDirectoryEndpoint, tenantID string, apiV
 		DeviceCodeEndpoint: *deviceCodeURL,
 	}, nil
 }
+
+// MultiTenantOAuthConfig provides endpoints for primary and aulixiary tenant IDs.
+type MultiTenantOAuthConfig interface {
+	PrimaryTenant() *OAuthConfig
+	AuxiliaryTenants() []*OAuthConfig
+}
+
+// Config contains optional OAuthConfig creation arguments.
+type Config struct {
+	APIVersion string
+}
+
+func (c Config) apiVersion() string {
+	if c.APIVersion != "" {
+		return fmt.Sprintf("?api-version=%s", c.APIVersion)
+	}
+	return "1.0"
+}
+
+// NewMultiTenantOAuthConfig creates an object that support multitenant OAuth configuration.
+func NewMultiTenantOAuthConfig(activeDirectoryEndpoint, primaryTenantID string, auxiliaryTenantIDs []string, cfg Config) (MultiTenantOAuthConfig, error) {
+	if len(auxiliaryTenantIDs) == 0 || len(auxiliaryTenantIDs) > 3 {
+		return nil, errors.New("must specify one to three auxiliary tenants")
```

Comment:
>three auxiliary ten [](start = 46, length = 19)

is there any documentation pointing to this ?

Reply:
Yes see https://docs.microsoft.com/en-us/azure/azure-resource-manager/authenticate-multi-tenant

Reply:
should we point in the comment to that page ?

---
In reply to: [297407597](https://github.com/Azure/go-autorest/pull/412#discussion_r297407597) [](ancestors = 297407597)

Reply:
Yep was thinking the same, will add.